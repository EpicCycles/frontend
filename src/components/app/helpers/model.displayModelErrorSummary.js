import {customerAddressFields} from "../model/helpers/fields";
import {displayModelErrorSummary} from "../model/helpers/model";

test('if no error or error_detail undefined returned', () => {
    const model = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6
    };
    const result = displayModelErrorSummary(model, customerAddressFields);
    expect(result).toBe(undefined);
});
test('if just error no error_detail error returned', () => {
    const model = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6,
        error: "Model level error"
    };
    const result = displayModelErrorSummary(model, customerAddressFields);
    expect(result).toBe("Model level error");
});
test('if no error with error_detail error_details formatted returned', () => {
    const model = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6,
        error_detail: {
            "address1": ["A valid address is required.", "SHow me the address is required."],
            "postcode": ["This field may not be blank."]
        },
    };
    const expectedText = "Address: A valid address is required. SHow me the address is required.<br>Postcode: This field may not be blank.";
    const result = displayModelErrorSummary(model, customerAddressFields);
    expect(result).toBe(expectedText);
});
test('if error and error_detail formatted returned', () => {
    const model = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6,
        error: "Model level error",
        error_detail: {
            "address1": ["A valid address is required.", "SHow me the address is required."],
            "postcode": ["This field may not be blank."]
        },
    };
    const expectedText = "Model level error<br>Address: A valid address is required. SHow me the address is required.<br>Postcode: This field may not be blank.";
    const result = displayModelErrorSummary(model, customerAddressFields);
    expect(result).toBe(expectedText);
});