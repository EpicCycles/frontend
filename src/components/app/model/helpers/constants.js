export const POSTCODE_RULES = [
  { countryCode: 'AT', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'AU', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'BE', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'BG', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'CA', regex: /^[A-Za-z][0-9][A-Za-z] [0-9][A-Za-z][0-9]/, display: 'ANA NAN' },
  { countryCode: 'CH', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'CY', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'CZ', regex: /^\d{3}\s\d{2}$/, display: 'NNN NN' },
  { countryCode: 'DE', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'DK', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'EE', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'ES', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'FI', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'FR', regex: /^\d{5}$/, display: 'NNNNN' },
  {
    countryCode: 'GB',
    regex: /^[A-Z](([0-9][0-9]?)|([A-Z][0-9][0-9]?)|([A-Z]?[0-9][A-Z])) ?[0-9][ABD-HJLNP-UW-Z]{2}$/,
    display: 'CCNN NCC',
  },
  { countryCode: 'GR', regex: /^\d{3}\s\d{2}$/, display: 'NNNNN or NNN NN' },
  { countryCode: 'HU', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'IT', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'KR', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'LT', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'LU', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'MT', regex: /^\[A-Z]{3}\s\d{4}$/, display: 'AAA NNNN' },
  { countryCode: 'MY', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'NO', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'NZ', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'PH', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'PL', regex: /^\d{2}-\d{3}$/, display: 'NN-NNN' },
  { countryCode: 'PT', regex: /^\d{4}-\d{3}$/, display: 'NNNN-NNN' },
  { countryCode: 'RO', regex: /^\d{6}$/, display: 'NNNNNN' },
  { countryCode: 'SE', regex: /^\d{3,4}$/, display: 'NNNNN or NNNN' },
  { countryCode: 'SG', regex: /^\d{6}$/, display: 'NNNNNN' },
  { countryCode: 'SI', regex: /^\d{3}\s\d{2}$/, display: 'NNN NN' },
  { countryCode: 'SK', regex: /^\d{3}\s\d{2}$/, display: 'NNN NN' },
  { countryCode: 'US', regex: /^(\d{5}([-]\d{4})?)$/, display: 'NNNNN-NNNN or NNNNN' },
  { countryCode: 'ZA', regex: /^\d{4}$/, display: 'NNNN' },
];
