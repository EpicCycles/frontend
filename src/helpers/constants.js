export const NEW_ELEMENT_ID = 'new';

export const colourStyles = [
  { background: 'bg-col-01', colour: 'col-01', border: 'border-col-01' },
  {
    background: 'bg-col-02',
    colour: 'col-02',
    border: 'border-col-02',
    transition: 'transition-01-02',
  },
  {
    background: 'bg-col-03',
    colour: 'col-03',
    border: 'border-col-03',
    transition: 'transition-02-03',
  },
  {
    background: 'bg-col-04',
    colour: 'col-04',
    border: 'border-col-04',
    transition: 'transition-03-04',
  },
  {
    background: 'bg-col-05',
    colour: 'col-05',
    border: 'border-col-05',
    transition: 'transition-04-05',
  },
  {
    background: 'bg-col-06',
    colour: 'col-06',
    border: 'border-col-06',
    transition: 'transition-05-06',
  },
  {
    background: 'bg-col-07',
    colour: 'col-07',
    border: 'border-col-07',
    transition: 'transition-06-07',
  },
  {
    background: 'bg-col-08',
    colour: 'col-08',
    border: 'border-col-08',
    transition: 'transition-07-08',
  },
  {
    background: 'bg-col-09',
    colour: 'col-09',
    border: 'border-col-09',
    transition: 'transition-08-09',
  },
  {
    background: 'bg-col-10',
    colour: 'col-10',
    border: 'border-col-10',
    transition: 'transition-09-10',
  },
  {
    background: 'bg-col-11',
    colour: 'col-11',
    border: 'border-col-11',
    transition: 'transition-10-11',
  },
  {
    background: 'bg-col-12',
    colour: 'col-12',
    border: 'border-col-12',
    transition: 'transition-11-12',
  },
  {
    background: 'bg-col-13',
    colour: 'col-13',
    border: 'border-col-13',
    transition: 'transition-12-13',
  },
];

export const STORAGE_PARTS = 'epic_parts';
export const STORAGE_SUPPLIER_PRODUCTS = 'epic_supplierProducts';
export const STORAGE_SECTIONS = 'epic_sections';
export const STORAGE_BRANDS = 'epic_brands';
export const STORAGE_CHARGES = 'epic_charges';
export const STORAGE_SUPPLIERS = 'epic_suppliers';
export const COOKIE_USER = 'epic_user';
export const COOKIE_TOKEN = 'epic_user_token';
