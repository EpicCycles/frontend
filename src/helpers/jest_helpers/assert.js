export const assertComponentHasExpectedProps = (component, expectedProps) =>
    Object.entries(expectedProps, ([propName, propValue]) => {
        if (propValue === 'function') {
            expect(
                component.prop(propName),
                `${propName} should be an instance of a function`
            ).to.be.instanceOf(Function);
            return;
        }
        expect(
            component.prop(propName),
            `${propName} should equal ${propValue}`
        ).to.eql(propValue);
    });

export const findDataTest = (component, dataTest) => {
    return component.find(`[data-test="${dataTest}"]`)
};
