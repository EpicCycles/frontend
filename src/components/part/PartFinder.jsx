import React from 'react'
import * as PropTypes from "prop-types";

import {findObjectWithKey, updateObject, updateObjectWithSelectionChanges} from "../../helpers/utils";
import Icon from "semantic-ui-react/dist/commonjs/elements/Icon/Icon";
import PartSelect from "./PartSelect";
import {partFieldsComplete} from "../app/model/helpers/fields";
import {addFieldToState} from "../app/model/helpers/model";
import {getModelFields, partReadyToUse} from "./helpers/part";
import EditModelPage from "../app/model/EditModelPage";
import PartSearch from "./PartSearch";

class PartFinder extends React.Component {
    componentWillMount() {
        this.setState(this.deriveStateFromProps());
    };

    deriveStateFromProps() {
        const { partType, part } = this.props;
        let partTypeSelected = partType;
        let brandSelected;
        if (part) {
            partTypeSelected = part.partType;
            brandSelected = part.brand;
        }
        this.setState({
            partTypeSelected,
            brandSelected,
            part,
            persistedPart: part,
        })
    }

    updateStateWithSelectionChanges = (fieldName, value) => {
        this.setState(updateObjectWithSelectionChanges(this.state, fieldName, value));
    };

    updatePartFieldsInState = (fieldName, value) => {
        const part = addFieldToState(this.state.part, partFieldsComplete, fieldName, value);
        this.updateStateWithSelectionChanges('part', part);
    };

    findParts = () => {
        const { partTypeSelected, brandSelected, searchPartName, searchStandard, searchStocked } = this.state;
        this.props.findParts({
            partType: partTypeSelected,
            partName: searchPartName,
            brand: brandSelected,
            standard: searchStandard,
            stocked: searchStocked,
        })
    };

    changePartViewed = (fieldName, partId) => {
        const part = findObjectWithKey(this.props.parts, partId);
        if (part) {
            if (this.state.part && this.state.part.changed) {
                if (window.confirm("You have made changes to the part, do you want to change the part and lose those changes?")) this.updateStateWithSelectionChanges('part', part);
            } else {
                this.updateStateWithSelectionChanges('part', part)
            }
        }
    };
    savePart = () => {
        this.props.savePart(this.state.part);
        this.updateStateWithSelectionChanges('persistedPart', this.state.part);
    };
    resetPart = () => {
        this.updateStateWithSelectionChanges('part', this.state.persistedPart);
    };
    deletePart = () => {
        const part = this.state.part;
        if (part.id) this.props.deletePart(part.id);
        this.setState(updateObject(this.state, { part: {}, persistedPart: {} }));
    };
    checkAndContinue = (nextAction) => {
        if (this.state.part && this.state.part.changed) {
            if (window.confirm("You have made changes to the part. If you want to keep them cancel and save the changes then continue.")) nextAction(this.state.part);
        } else {
            nextAction(this.state.part);
        }
    };

    render() {
        const { partTypeSelected, brandSelected, searchPartName, searchStandard, searchStocked, part, persistedPart } = this.state;
        const { closeAction, sections, brands, parts, partActionPrimary, partActionPrimaryIcon, partActionPrimaryTitle, partActionSecondary, partActionSecondaryIcon, partActionSecondaryTitle } = this.props;

        let partsForSelect = part ? parts.filter(part => part.partType === partTypeSelected) : [];
        if (brandSelected) partsForSelect = partsForSelect.filter(part => part.brand === brandSelected);
        if (searchStandard) partsForSelect = partsForSelect.filter(part => part.standard);
        if (searchStocked) partsForSelect = partsForSelect.filter(part => part.stocked);
        return <div className="grid-container">
            <h2>Find Part</h2>
            {closeAction && <div style={{ width: "100%", textAlign: "right" }}>
                <Icon
                    name="remove"
                    circular
                    link
                    onClick={() => this.checkAndContinue(closeAction)}
                />
            </div>}
            <div key="partFinderFields" className={`grid`}>
                <PartSearch
                    brands={brands}
                    sections={sections}
                    onChange={this.updateStateWithSelectionChanges}
                    findParts={this.findParts}
                    partTypeSelected={partTypeSelected}
                    brandSelected={brandSelected}
                    searchPartName={searchPartName}
                    searchStandard={searchStandard}
                    searchStocked={searchStocked}
                />
                {(partsForSelect.length > 0) && <div
                    className="grid-row"
                    key={`runSearchRow`}
                >
                    <div
                        className="grid-item--borderless field-label align_right"
                        key={`selectStandard`}
                    >
                        Choose Part:
                    </div>
                    <div
                        className="grid-item--borderless field-label "
                    >
                        <PartSelect
                            fieldName="partId"
                            brands={brands}
                            parts={partsForSelect}
                            onChange={this.changePartViewed}
                            partSelected={part}
                        />
                    </div>
                </div>}
            </div>
            <EditModelPage
                modelFields={getModelFields(part, (!(part && part.id)))}
                model={part}
                persistedModel={persistedPart}
                sections={sections}
                brands={brands}
                onChange={this.updatePartFieldsInState}
            />
            {part && <div style={{ width: "100%", textAlign: "right" }}>
                {part.changed &&
                <Icon id={`reset-part`} name="undo"
                      onClick={this.resetPart}
                      title="Reset Part details"
                />
                }
                {partReadyToUse(part) &&
                <Icon
                    key="primaryAction"
                    name={partActionPrimaryIcon}
                    onClick={() => partActionPrimary(part)}
                    title={partActionPrimaryTitle}
                />
                }
                {partActionSecondary &&
                <Icon
                    key="secondaryAction"
                    name={partActionSecondaryIcon}
                    onClick={() => partActionSecondary(part)}
                    title={partActionSecondaryTitle}
                />
                }
            </div>}
        </div>
    }
}

PartFinder.defaultProps = {
    parts: [],
};
PartFinder.propTypes = {
    sections: PropTypes.array.isRequired,
    brands: PropTypes.array.isRequired,
    parts: PropTypes.array,
    partType: PropTypes.number,
    part: PropTypes.object,
    findParts: PropTypes.func.isRequired,
    closeAction: PropTypes.func.isRequired,
    partActionPrimary: PropTypes.func.isRequired,
    partActionPrimaryIcon: PropTypes.string.isRequired,
    partActionPrimaryTitle: PropTypes.string.isRequired,
    partActionSecondary: PropTypes.func,
    partActionSecondaryIcon: PropTypes.string,
    partActionSecondaryTitle: PropTypes.string,
};

export default PartFinder;