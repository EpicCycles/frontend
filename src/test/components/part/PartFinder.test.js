import React from 'react';
import toJson from 'enzyme-to-json';
import PartFinder from '../../../components/part/PartFinder';
import { NEW_ELEMENT_ID } from '../../../helpers/constants';
import { sampleSections } from '../../../helpers/sampleData';
import { assertComponentHasExpectedProps } from '../../jest_helpers/assert';

const brands = [
  { id: 1, brand_name: 'brand 1' },
  { id: 2, brand_name: 'brand 2', supplier: [], supplier_names: [] },
  { id: 3, brand_name: 'brand 3', supplier: [1], supplier_names: ['supplier 1'] },
  { id: 4, brand_name: 'brand 4', delete: true },
  {
    id: 5,
    brand_name: 'brand 5',
    changed: true,
    supplier: [1, 3],
    supplier_names: ['supplier 1', 'supplier 3'],
  },
  { dummyKey: '123ABC', brand_name: 'brand new', changed: true },
  { dummyKey: NEW_ELEMENT_ID, brand_name: 'brand new 2', changed: true },
];

const parts = [
  {
    id: 65,
    part_name: 'A-Head Tapered Cartridge aluminium',
    trade_in_price: null,
    standard: false,
    stocked: false,
    partType: 3,
    brand: 3,
  },
  {
    id: 329,
    brand_name: 'Shimano',
    part_name: 'SLX M7000 rapidfire',
    trade_in_price: null,
    standard: false,
    stocked: false,
    partType: 5,
    brand: 6,
  },
  {
    id: 330,
    brand_name: 'Shimano',
    part_name: 'Deore XT M8000 Shadow Plus 11 speed',
    trade_in_price: null,
    standard: false,
    stocked: false,
    partType: 7,
    brand: 6,
  },
];

test('should render with minimum information', () => {
  const props = {
    sections: sampleSections,
    brands,
    savePart: jest.fn(),
    deletePart: jest.fn(),
    closeAction: jest.fn(),
    findParts: jest.fn(),
    partActionPrimary: jest.fn(),
    partActionPrimaryIcon: 'meh',
    partActionPrimaryTitle: 'select',
  };
  const component = shallow(<PartFinder {...props} />);
  expect(toJson(component)).toMatchSnapshot();
  const PartTypeSelect = component.find('PartTypeSelect');
  assertComponentHasExpectedProps(PartTypeSelect, {
    sections: props.sections,
    isEmptyAllowed: true,
  });
  const BrandSelect = component.find('BrandSelect');
  assertComponentHasExpectedProps(BrandSelect, {
    brands: props.brands,
    isEmptyAllowed: true,
  });
});
test('should render with maximum information', () => {
  const props = {
    sections: sampleSections,
    brands,
    parts,
    part: parts[2],
    partType: parts[2].partType,
    closeAction: jest.fn(),
    savePart: jest.fn(),
    deletePart: jest.fn(),
    findParts: jest.fn(),
    partActionPrimary: jest.fn(),
    partActionPrimaryIcon: 'meh',
    partActionPrimaryTitle: 'select',
    partActionSecondary: jest.fn(),
    partActionSecondaryIcon: 'delete',
    partActionSecondaryTitle: 'Delete me',
  };
  const component = shallow(<PartFinder {...props} />);
  expect(toJson(component)).toMatchSnapshot();
  const PartTypeSelect = component.find('PartTypeSelect');
  assertComponentHasExpectedProps(PartTypeSelect, {
    sections: props.sections,
    isEmptyAllowed: true,
  });
  const BrandSelect = component.find('BrandSelect');
  assertComponentHasExpectedProps(BrandSelect, {
    brands: props.brands,
    isEmptyAllowed: true,
  });
  const PartSelect = component.find('PartSelect');
  assertComponentHasExpectedProps(PartSelect, {
    parts: props.parts,
    partSelected: parts[2],
  });
});
