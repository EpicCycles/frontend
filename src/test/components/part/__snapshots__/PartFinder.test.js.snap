// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with maximum information 1`] = `
<div
  className="grid-container"
>
  <Prompt
    message="You have made changes. Cancel and Save if you do not want to lose them."
  />
  <h2>
    Find Part
  </h2>
  <div
    style={
      Object {
        "textAlign": "right",
        "width": "100%",
      }
    }
  >
    <Icon
      as="i"
      circular={true}
      link={true}
      name="remove"
      onClick={[Function]}
    />
  </div>
  <div
    className="grid"
    key="partFinderFields"
  >
    <PartSearch
      brands={
        Array [
          Object {
            "brand_name": "brand 1",
            "id": 1,
          },
          Object {
            "brand_name": "brand 2",
            "id": 2,
            "supplier": Array [],
          },
          Object {
            "brand_name": "brand 3",
            "id": 3,
            "supplier": Array [
              1,
            ],
          },
          Object {
            "brand_name": "brand 4",
            "delete": true,
            "id": 4,
          },
          Object {
            "brand_name": "brand 5",
            "changed": true,
            "id": 5,
            "supplier": Array [
              1,
              3,
            ],
          },
          Object {
            "brand_name": "brand new",
            "changed": true,
            "dummyKey": "123ABC",
          },
          Object {
            "brand_name": "brand new 2",
            "changed": true,
            "dummyKey": "new",
          },
        ]
      }
      findParts={[Function]}
      onChange={[Function]}
      searchCriteria={
        Object {
          "brandSelected": 6,
          "partTypeSelected": 7,
          "searchPartName": undefined,
          "searchStandard": undefined,
          "searchStocked": undefined,
        }
      }
      searchFields={
        Array [
          Object {
            "displayName": "Part Type:",
            "fieldName": "partTypeSelected",
            "modelFieldName": "partType",
            "selectList": Array [
              Object {
                "name": "Frame (Frameset)",
                "value": "1",
              },
              Object {
                "name": "Fork (Frameset)",
                "value": "2",
              },
              Object {
                "name": "Headset (Frameset)",
                "value": "3",
              },
            ],
            "type": "select_one",
          },
          Object {
            "displayName": "Brand:",
            "fieldName": "brandSelected",
            "modelFieldName": "brand",
            "selectList": Array [
              Object {
                "name": "brand 1",
                "value": "1",
              },
              Object {
                "name": "brand 2",
                "value": "2",
              },
              Object {
                "name": "brand 3",
                "value": "3",
              },
              Object {
                "name": "brand 4",
                "value": "4",
              },
              Object {
                "name": "brand 5",
                "value": "5",
              },
              Object {
                "name": "brand new",
                "value": "123ABC",
              },
              Object {
                "name": "brand new 2",
                "value": "new",
              },
            ],
            "type": "select_one",
          },
          Object {
            "displayName": "Part name contains:",
            "fieldName": "searchPartName",
            "modelFieldName": "part_name",
            "placeholder": "part name",
            "type": "text",
          },
          Object {
            "displayName": "Standard only",
            "fieldName": "searchStandard",
            "modelFieldName": "standard",
            "type": "checkbox",
          },
          Object {
            "displayName": "Stocked only",
            "fieldName": "searchStocked",
            "modelFieldName": "stocked",
            "type": "checkbox",
          },
        ]
      }
      sections={
        Array [
          Object {
            "id": 1,
            "name": "Frameset",
            "partTypes": Array [
              Object {
                "attributes": Array [
                  Object {
                    "attribute_name": "Size",
                    "attribute_type": "1",
                    "id": 1,
                    "in_use": true,
                    "mandatory": true,
                    "options": Array [],
                    "partType": 1,
                    "placing": 10,
                  },
                ],
                "can_be_omitted": false,
                "can_be_substituted": true,
                "customer_facing": false,
                "id": 1,
                "includeInSection": 1,
                "name": "Frame",
                "placing": 10,
                "synonyms": Array [],
              },
              Object {
                "attributes": Array [],
                "can_be_omitted": true,
                "can_be_substituted": false,
                "customer_facing": false,
                "id": 2,
                "includeInSection": 1,
                "name": "Fork",
                "placing": 20,
                "synonyms": Array [
                  Object {
                    "id": 1,
                    "name": "Front Fork",
                    "partType": 2,
                  },
                ],
              },
              Object {
                "attributes": Array [],
                "can_be_omitted": false,
                "can_be_substituted": false,
                "customer_facing": true,
                "id": 3,
                "includeInSection": 1,
                "name": "Headset",
                "placing": 30,
                "synonyms": Array [],
              },
            ],
            "placing": 10,
          },
        ]
      }
    />
  </div>
  <EditModelPage
    brands={
      Array [
        Object {
          "brand_name": "brand 1",
          "id": 1,
        },
        Object {
          "brand_name": "brand 2",
          "id": 2,
          "supplier": Array [],
        },
        Object {
          "brand_name": "brand 3",
          "id": 3,
          "supplier": Array [
            1,
          ],
        },
        Object {
          "brand_name": "brand 4",
          "delete": true,
          "id": 4,
        },
        Object {
          "brand_name": "brand 5",
          "changed": true,
          "id": 5,
          "supplier": Array [
            1,
            3,
          ],
        },
        Object {
          "brand_name": "brand new",
          "changed": true,
          "dummyKey": "123ABC",
        },
        Object {
          "brand_name": "brand new 2",
          "changed": true,
          "dummyKey": "new",
        },
      ]
    }
    model={
      Object {
        "brand": 6,
        "brand_name": "Shimano",
        "id": 330,
        "partType": 7,
        "part_name": "Deore XT M8000 Shadow Plus 11 speed",
        "standard": false,
        "stocked": false,
        "trade_in_price": null,
      }
    }
    modelFields={
      Array [
        Object {
          "error": "A brand must be selected",
          "fieldName": "brand",
          "header": "Brand",
          "required": true,
          "synonyms": Array [
            "brand",
            "manufacturer",
          ],
          "type": "brand",
        },
        Object {
          "error": "A name is required for the part",
          "fieldName": "part_name",
          "header": "Part Name",
          "length": 200,
          "maxWidth": "250px",
          "required": true,
          "size": 40,
        },
        Object {
          "fieldName": "trade_in_price",
          "header": "Trade In Â£",
          "length": 10,
          "size": 10,
          "type": "currency",
        },
        Object {
          "fieldName": "standard",
          "header": "Standard",
          "type": "checkbox",
        },
      ]
    }
    onChange={[Function]}
    persistedModel={
      Object {
        "brand": 6,
        "brand_name": "Shimano",
        "id": 330,
        "partType": 7,
        "part_name": "Deore XT M8000 Shadow Plus 11 speed",
        "standard": false,
        "stocked": false,
        "trade_in_price": null,
      }
    }
    sections={
      Array [
        Object {
          "id": 1,
          "name": "Frameset",
          "partTypes": Array [
            Object {
              "attributes": Array [
                Object {
                  "attribute_name": "Size",
                  "attribute_type": "1",
                  "id": 1,
                  "in_use": true,
                  "mandatory": true,
                  "options": Array [],
                  "partType": 1,
                  "placing": 10,
                },
              ],
              "can_be_omitted": false,
              "can_be_substituted": true,
              "customer_facing": false,
              "id": 1,
              "includeInSection": 1,
              "name": "Frame",
              "placing": 10,
              "synonyms": Array [],
            },
            Object {
              "attributes": Array [],
              "can_be_omitted": true,
              "can_be_substituted": false,
              "customer_facing": false,
              "id": 2,
              "includeInSection": 1,
              "name": "Fork",
              "placing": 20,
              "synonyms": Array [
                Object {
                  "id": 1,
                  "name": "Front Fork",
                  "partType": 2,
                },
              ],
            },
            Object {
              "attributes": Array [],
              "can_be_omitted": false,
              "can_be_substituted": false,
              "customer_facing": true,
              "id": 3,
              "includeInSection": 1,
              "name": "Headset",
              "placing": 30,
              "synonyms": Array [],
            },
          ],
          "placing": 10,
        },
      ]
    }
  />
  <div
    style={
      Object {
        "textAlign": "right",
        "width": "100%",
      }
    }
  >
    <Icon
      as="i"
      key="primaryAction"
      name="meh"
      onClick={[Function]}
      title="select"
    />
    <Icon
      as="i"
      key="secondaryAction"
      name="delete"
      onClick={[Function]}
      title="Delete me"
    />
  </div>
</div>
`;

exports[`should render with minimum information 1`] = `
<div
  className="grid-container"
>
  <Prompt
    message="You have made changes. Cancel and Save if you do not want to lose them."
  />
  <h2>
    Find Part
  </h2>
  <div
    style={
      Object {
        "textAlign": "right",
        "width": "100%",
      }
    }
  >
    <Icon
      as="i"
      circular={true}
      link={true}
      name="remove"
      onClick={[Function]}
    />
  </div>
  <div
    className="grid"
    key="partFinderFields"
  >
    <PartSearch
      brands={
        Array [
          Object {
            "brand_name": "brand 1",
            "id": 1,
          },
          Object {
            "brand_name": "brand 2",
            "id": 2,
            "supplier": Array [],
          },
          Object {
            "brand_name": "brand 3",
            "id": 3,
            "supplier": Array [
              1,
            ],
          },
          Object {
            "brand_name": "brand 4",
            "delete": true,
            "id": 4,
          },
          Object {
            "brand_name": "brand 5",
            "changed": true,
            "id": 5,
            "supplier": Array [
              1,
              3,
            ],
          },
          Object {
            "brand_name": "brand new",
            "changed": true,
            "dummyKey": "123ABC",
          },
          Object {
            "brand_name": "brand new 2",
            "changed": true,
            "dummyKey": "new",
          },
        ]
      }
      findParts={[Function]}
      onChange={[Function]}
      searchCriteria={
        Object {
          "brandSelected": undefined,
          "partTypeSelected": undefined,
          "searchPartName": undefined,
          "searchStandard": undefined,
          "searchStocked": undefined,
        }
      }
      searchFields={
        Array [
          Object {
            "displayName": "Part Type:",
            "fieldName": "partTypeSelected",
            "modelFieldName": "partType",
            "selectList": Array [
              Object {
                "name": "Frame (Frameset)",
                "value": "1",
              },
              Object {
                "name": "Fork (Frameset)",
                "value": "2",
              },
              Object {
                "name": "Headset (Frameset)",
                "value": "3",
              },
            ],
            "type": "select_one",
          },
          Object {
            "displayName": "Brand:",
            "fieldName": "brandSelected",
            "modelFieldName": "brand",
            "selectList": Array [
              Object {
                "name": "brand 1",
                "value": "1",
              },
              Object {
                "name": "brand 2",
                "value": "2",
              },
              Object {
                "name": "brand 3",
                "value": "3",
              },
              Object {
                "name": "brand 4",
                "value": "4",
              },
              Object {
                "name": "brand 5",
                "value": "5",
              },
              Object {
                "name": "brand new",
                "value": "123ABC",
              },
              Object {
                "name": "brand new 2",
                "value": "new",
              },
            ],
            "type": "select_one",
          },
          Object {
            "displayName": "Part name contains:",
            "fieldName": "searchPartName",
            "modelFieldName": "part_name",
            "placeholder": "part name",
            "type": "text",
          },
          Object {
            "displayName": "Standard only",
            "fieldName": "searchStandard",
            "modelFieldName": "standard",
            "type": "checkbox",
          },
          Object {
            "displayName": "Stocked only",
            "fieldName": "searchStocked",
            "modelFieldName": "stocked",
            "type": "checkbox",
          },
        ]
      }
      sections={
        Array [
          Object {
            "id": 1,
            "name": "Frameset",
            "partTypes": Array [
              Object {
                "attributes": Array [
                  Object {
                    "attribute_name": "Size",
                    "attribute_type": "1",
                    "id": 1,
                    "in_use": true,
                    "mandatory": true,
                    "options": Array [],
                    "partType": 1,
                    "placing": 10,
                  },
                ],
                "can_be_omitted": false,
                "can_be_substituted": true,
                "customer_facing": false,
                "id": 1,
                "includeInSection": 1,
                "name": "Frame",
                "placing": 10,
                "synonyms": Array [],
              },
              Object {
                "attributes": Array [],
                "can_be_omitted": true,
                "can_be_substituted": false,
                "customer_facing": false,
                "id": 2,
                "includeInSection": 1,
                "name": "Fork",
                "placing": 20,
                "synonyms": Array [
                  Object {
                    "id": 1,
                    "name": "Front Fork",
                    "partType": 2,
                  },
                ],
              },
              Object {
                "attributes": Array [],
                "can_be_omitted": false,
                "can_be_substituted": false,
                "customer_facing": true,
                "id": 3,
                "includeInSection": 1,
                "name": "Headset",
                "placing": 30,
                "synonyms": Array [],
              },
            ],
            "placing": 10,
          },
        ]
      }
    />
  </div>
  <EditModelPage
    brands={
      Array [
        Object {
          "brand_name": "brand 1",
          "id": 1,
        },
        Object {
          "brand_name": "brand 2",
          "id": 2,
          "supplier": Array [],
        },
        Object {
          "brand_name": "brand 3",
          "id": 3,
          "supplier": Array [
            1,
          ],
        },
        Object {
          "brand_name": "brand 4",
          "delete": true,
          "id": 4,
        },
        Object {
          "brand_name": "brand 5",
          "changed": true,
          "id": 5,
          "supplier": Array [
            1,
            3,
          ],
        },
        Object {
          "brand_name": "brand new",
          "changed": true,
          "dummyKey": "123ABC",
        },
        Object {
          "brand_name": "brand new 2",
          "changed": true,
          "dummyKey": "new",
        },
      ]
    }
    model={Object {}}
    modelFields={
      Array [
        Object {
          "error": "A part type must be selected",
          "fieldName": "partType",
          "header": "Part Type",
          "required": true,
          "type": "partType",
        },
        Object {
          "error": "A brand must be selected",
          "fieldName": "brand",
          "header": "Brand",
          "required": true,
          "synonyms": Array [
            "brand",
            "manufacturer",
          ],
          "type": "brand",
        },
        Object {
          "error": "A name is required for the part",
          "fieldName": "part_name",
          "header": "Part Name",
          "length": 200,
          "maxWidth": "250px",
          "required": true,
          "size": 40,
        },
        Object {
          "fieldName": "trade_in_price",
          "header": "Trade In Â£",
          "length": 10,
          "size": 10,
          "type": "currency",
        },
        Object {
          "fieldName": "standard",
          "header": "Standard",
          "type": "checkbox",
        },
      ]
    }
    onChange={[Function]}
    persistedModel={Object {}}
    sections={
      Array [
        Object {
          "id": 1,
          "name": "Frameset",
          "partTypes": Array [
            Object {
              "attributes": Array [
                Object {
                  "attribute_name": "Size",
                  "attribute_type": "1",
                  "id": 1,
                  "in_use": true,
                  "mandatory": true,
                  "options": Array [],
                  "partType": 1,
                  "placing": 10,
                },
              ],
              "can_be_omitted": false,
              "can_be_substituted": true,
              "customer_facing": false,
              "id": 1,
              "includeInSection": 1,
              "name": "Frame",
              "placing": 10,
              "synonyms": Array [],
            },
            Object {
              "attributes": Array [],
              "can_be_omitted": true,
              "can_be_substituted": false,
              "customer_facing": false,
              "id": 2,
              "includeInSection": 1,
              "name": "Fork",
              "placing": 20,
              "synonyms": Array [
                Object {
                  "id": 1,
                  "name": "Front Fork",
                  "partType": 2,
                },
              ],
            },
            Object {
              "attributes": Array [],
              "can_be_omitted": false,
              "can_be_substituted": false,
              "customer_facing": true,
              "id": 3,
              "includeInSection": 1,
              "name": "Headset",
              "placing": 30,
              "synonyms": Array [],
            },
          ],
          "placing": 10,
        },
      ]
    }
  />
  <div
    style={
      Object {
        "textAlign": "right",
        "width": "100%",
      }
    }
  />
</div>
`;
